testhelpers/consensus.go

```diff
@@ -203,6 +218,53 @@ func (mbv *StubBlockValidator) StubSemanticValidationForBlock(child *types.Block
 	mbv.semanticStubs[child.Cid()] = err
 }
 
+// FakeElectionMachine generates fake election proofs and verifies all proofs
+type FakeElectionMachine struct{}
+
+// RunElection returns a fake election proof.
+func (fem *FakeElectionMachine) RunElection(ticket types.Ticket, candidateAddr address.Address, signer consensus.TicketSigner) (types.VRFPi, error) {
+	return MakeFakeElectionProofForTest(), nil
+}
+
+// IsElectionWinner always returns true
+func (fem *FakeElectionMachine) IsElectionWinner(ctx context.Context, bs blockstore.Blockstore, ptv consensus.PowerTableView, st state.Tree, ticket types.Ticket, electionProof types.VRFPi, signerAddr, minerAddr address.Address) (bool, error) {
+	return true, nil
+}
+
+// FakeTicketMachine generates fake tickets and verifies all tickets
+type FakeTicketMachine struct{}
+
+// NextTicket returns a fake ticket
+func (ftm *FakeTicketMachine) NextTicket(parent types.Ticket, signerAddr address.Address, signer consensus.TicketSigner, nullBlkCount uint64) (types.Ticket, error) {
+	return MakeFakeTicketForTest(), nil
+}
+
+// NotarizeTime does nothing
+func (ftm *FakeTicketMachine) NotarizeTime(ticket *types.Ticket) error {
+	return nil
+}
+
+// ValidateTicket always returns true
+func (ftm *FakeTicketMachine) ValidateTicket(parent, ticket types.Ticket, signerAddr address.Address, nullBlkCount uint64) (bool, error) {
+	return true, nil
+}
+
+// FailingTicketValidator marks all tickets as invalid
+type FailingTicketValidator struct{}
+
+// ValidateTicket always returns false
+func (ftv *FailingTicketValidator) ValidateTicket(parent, ticket types.Ticket, signerAddr address.Address, nullBlkCount uint64) (bool, error) {
+	return false, nil
+}
+
+// FailingElectionValidator marks all elections as invalid
+type FailingElectionValidator struct{}
+
+// IsElectionWinner always returns false
+func (fev *FailingElectionValidator) IsElectionWinner(ctx context.Context, bs blockstore.Blockstore, ptv consensus.PowerTableView, st state.Tree, ticket types.Ticket, electionProof types.VRFPi, signerAddr, minerAddr address.Address) (bool, error) {
+	return false, nil
+}
+
```

Comment:
`testhelpers` is a dumping ground, I don't want new stuff here. Put test helpers alongside the relevant code, in `consensus/testing.go`, or `consensus/testing_xxx.go` to split things up.